### ISSUE TITLE
[sqllab] Add configuration option to disable data preview query in SQL editor

### ISSUE THREAD

**Issue by akedrou** [apache/superset#14726]
**Is your feature request related to a problem? Please describe.**
We are using Superset with Trino over very wide tables with complex nested data columns. When writing queries in SQL editor, `getTableMetadata` will automatically [dispatch a data preview query](https://github.com/apache/superset/blob/a9d888ad402ebb35da45df446997c426d6abee9d/superset-frontend/src/SqlLab/actions/sqlLab.js#L987-L1014) that can lock up the UI for a few seconds when it renders the results.

**Describe the solution you'd like**
I would like a configuration option to disable the data preview query. This setting could be global, but I think it makes most sense as a database `extra` configuration option `sql_editor_preview_data` (default to current behavior/`True`) that is returned in the `get_table_metadata` API response as `previewData`. If `False`, the query would be skipped.

**Describe alternatives you've considered**
 - Improving the performance of the response rendering (e.g. limiting the number of columns; ensuring the UI doesn't lock)
     - Limiting the number of columns defeats the purpose of a preview query.
     - Showing all the columns in such a wide table is not an effective preview anyway.
 - Placing the setting at a table/schema level
     - Would be nice to enable previews as much as possible, and this could be managed by table with something like a dictionary of table->bool
     - The benefit doesn't seem worth the cost of a confusing database setting.

**Additional context**
I'm happy to submit a PR for the feature assuming the proposed solution (or something similarly straightforward) works.

**Comment by junlincc** [comment]
i agree, we currently send out a preview query each time when selecting a table... it  can be cumbersome for data warehouse like clickhouse with maaaaany columns. please do open a PR. :) thanks for suggesting 

cc @betodealmeida @eschutho

**Comment by eschutho** [comment]
@yousoph

**Comment by akedrou** [comment]
Hi @junlincc - I opened a PR for this change a few weeks back (#14768). Sorry it took a bit to follow up, but is there some other step I need to follow before tests are approved to run on it and it gets reviewed or is it just waiting on someone to have some time? I checked the guide for contributing but it's entirely possible I missed something. Thanks!
